tiledb/sm/array_schema/test/array_schema_test_support.h

```diff
@@ -0,0 +1,333 @@
+/**
+ * @file tiledb/sm/array_schema/test/array_schema_test_support.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This header defines support functions for defining array schemas and their
+ * component objects.
+ *
+ * The array schema classes have constructors that operate only at a low level,
+ * with arguments identical to (or at least very close to) the member variables.
+ * As a result manual construction of array schemas is rather verbose when the
+ * purpose is to define particular schemas rather than to support persistence to
+ * storage or to implement the C API.
+ *
+ * The overall approach here is to wrap the schema classes to allow more compact
+ * construction patterns. Using classes rather than factory functions means that
+ * all the details of the underlying schema constructors is hidden during
+ * construction.
+ *
+ * It is a top-level goal of this design that the constructors for an array
+ * schema stand alone, that they not require construction of component parts
+ * beforehand. These constructors avoid declaration junk, the separate variables
+ * that might not be referenced later. Each of the schema components are also
+ * constructible separately for testing situations where an entire array schema
+ * is not required.
+ *
+ * These classes are "inside the API"; they don't call through the C API in
```

Comment:
This is understandable for unit testing, but presents a very real risk of leading to incomplete coverage of the C and C++ APIs.